articles/customization-using-bulma/index.md

```diff
@@ -0,0 +1,234 @@
+title: Customize CSS using Bulma
+description: Bulma is a lightweight, open source CSS framework that has excellent customization features which is perfect for rapid development.
+author: saiharsha-balasubramaniam
+date: 2020-09-04T00:00:00-08:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/customization-using-bulma/hero.jpg
+    alt: bulma
+---
+
+**Cascading Style Sheets** is a stylesheet language that is used to describe how a webpage looks and feels visually. To speed up the development of projects, developers use CSS frameworks like Bulma. Let us see why CSS frameworks are useful and learn about Bulma.
+
+## Table of Contents
+
+- [The Need for a CSS Framework](#need-for-a-css-framework)
+- [Bulma](#bulma)
+- [Getting Started](#getting-started)
+- [Column System](#column-system)
+- [Components](#components)
+- [Customizing Variables](#customizing-variables)
+- [Further Reading](#further-reading)
+
+## The Need for a CSS Framework
+
+CSS is an flexible language, with a ton of customization options. This may sometimes be overwhelming for a beginner. When you're out to build a product, a CSS framework helps to accelerate the process of development. It helps to reduce time to production, which is crucial for a project. Some other benefits to using a CSS framework include:
+
+- Good CSS frameworks have excellent documentation.
+- Easier to make websites responsive.
+- Cross-browser support.
+
+## Bulma
+
+Bulma is a fast, lightweight and customizable CSS framework. Bulma is perfect for production-grade web applications due to its excellent performance and small size.
+
+- Bulma is based on CSS Flexbox.
+- It has an excellent developer community.
+- It can be customized extensively through variables.
+- Bulma uses easy-to-read class names that are easy to remember.
+- It provides a modern and clean design.
+
+For the above reasons, Bulma is an efficient CSS framework that can be used to create performant websites.
+
+## Getting Started
+
+Let's add Bulma to our web application. We can do this by one of the following ways.
+
+- We can use the Bulma CDN to add Bulma to our webpage. To do so in the website's `<head>`, add the following line.
+
+```html
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.0/css/bulma.min.css"/>
+```
+
+- Download the npm package from the npm repository.
+
+```bash
+npm install bulma
+```
+
+- We can also download the `bulma.css` file from [here](https://github.com/jgthms/bulma/tree/master/css) and include it in our webpage using the `<link>` tag.
+
+```html
+<link rel="stylesheet" href="bulma.css" />
+```
+
+## Column System
+
+One of the best features of Bulma is the responsive column grid system. For building a columns layout,
+
+- Every row in Bulma is defined by a `columns` container.
+- Every column should be defined as a `column` and should be nested within the `columns` container.
+
+For example:
+
+```html
+<div class="columns">
```

Comment:
This did not work for me using the NPM version of Bulma.

Reply:
> This did not work for me using the NPM version of Bulma.

Did you import the CSS? by running,
```css
import 'bulma/css/bulma.css'
```

Reply:
I found out why it wasn't working, @cyberShaw. The line I gave you (which is in Bulma's documentation as how you should import the NPM module) only works if you're using a bundler. I've given you an updated line of code which works without one.